samples/99_zil_interpreter/app/builtins.rb

```diff
@@ -262,3 +262,27 @@ def expect_minimum_argument_count!(arguments, min_count)
   false
 }
 
+ZIL_BUILTINS[:OBJECT] = lambda { |arguments, context|
+  # Objects have a name, and a list of properties and values
+
+  raise FunctionError, "OBJECT requires at least a name!" unless arguments.length > 0
```

Comment:
you could use my fancy helpers :wink: `expect_minimum_argument_count!(arguments, 1)` in this case

Reply:
+1 to this.... :)